element-{web,desktop}: 1.12.3 -> 1.12.6

teutat3s opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->


Release notes `element-web`:
- https://github.com/element-hq/element-web/releases/tag/v1.12.4
- https://github.com/element-hq/element-web/releases/tag/v1.12.5
- https://github.com/element-hq/element-web/releases/tag/v1.12.6

Changelog `element-web`:
- https://github.com/element-hq/element-web/compare/v1.12.3...v1.12.6

Release notes `element-desktop`:
- https://github.com/element-hq/element-desktop/releases/tag/v1.12.4
- https://github.com/element-hq/element-desktop/releases/tag/v1.12.5
- https://github.com/element-hq/element-desktop/releases/tag/v1.12.6

Changelog `element-desktop`:
- https://github.com/element-hq/element-desktop/compare/v1.12.3...v1.12.6

Needs a fix to add the `shared-components` dependency since https://github.com/element-hq/element-web/commit/e883b05206129857aa00ca726252e10a0eb05cf9 introduced a `link:` dependency that we need to fetch as well.
`fetchYarnDeps` in `fixup.js` only supports `file:` dependencies.
See: https://github.com/NixOS/nixpkgs/blob/7c05114ab1077b3e4309129c6bdfd416bb309890/pkgs/build-support/node/fetch-yarn-deps/fixup.js#L18

Co-authored-by: @akshaymankar

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

teutat3s committed:
element-web-unwrapped: 1.12.3 -> 1.12.4

Release notes: https://github.com/element-hq/element-web/releases/tag/v1.12.4
Changelog: https://github.com/element-hq/element-web/compare/v1.12.3...v1.12.4

teutat3s committed:
element-desktop: 1.12.3 -> 1.12.4

Release notes: https://github.com/element-hq/element-desktop/releases/tag/v1.12.4
Changelog: https://github.com/element-hq/element-desktop/compare/v1.12.3...v1.12.4

akshaymankar committed:
element-web-unwrapped: Build shared-components

- add comment for offlineCacheSharedComponents
- add webSharedComponentsYarnHash to update.sh

teutat3s commented:
```
error: hash mismatch in fixed-output derivation '/nix/store/qkblr82h95prg6b798xq0y9h6wdqh7v7-element-desktop-yarn-deps-1.12.4.drv':
specified: sha256-cDykxMKErempq+P3upXi13gm/8KJUqKfrk5jZaJECnQ=
got: sha256-7eYmz298m6ceREcaAo9CSMMFC3fB/ZC1+ArPb9pqc0o=
```
... I will fix it later today.

freshprince commented on pkgs/by-name/el/element-web-unwrapped/package.nix:
```diff
@@ -50,6 +57,12 @@ stdenv.mkDerivation (
       export VERSION=${finalAttrs.version}
+
+      pushd packages/shared-components
+      yarnOfflineCache=${offlineCacheSharedComponents} yarnConfigHook
```
Could you maybe use `finalAttrs.offlineCacheSharedComponents` so that this can be overriden?

teutat3s commented:
I'll update to 1.12.5 later...
https://github.com/element-hq/element-web/releases/tag/v1.12.5

teutat3s committed:
element-web-unwrapped: 1.12.4 -> 1.12.6

Release notes: https://github.com/element-hq/element-web/releases/tag/v1.12.5
Changelog: https://github.com/element-hq/element-web/compare/v1.12.4...v1.12.6

teutat3s committed:
element-desktop: 1.12.4 -> 1.12.6

Release notes: https://github.com/element-hq/element-desktop/releases/tag/v1.12.5
Changelog: https://github.com/element-hq/element-desktop/compare/v1.12.4...v1.12.6
